ui-speedspacechart/src/components/helpers/drawElements/declivity.ts

```diff
@@ -0,0 +1,62 @@
+import { clearCanvas, slopesValues, maxPositionValues } from '../../utils';
+import type { Store } from '../../../types/chartTypes';
+import { MARGINS } from '../../const';
+
+const { CURVE_MARGIN_SIDES, MARGIN_TOP, MARGIN_BOTTOM } = MARGINS;
+
+export const drawDeclivity = (
+  ctx: CanvasRenderingContext2D,
+  width: number,
+  height: number,
+  store: Store
+) => {
+  const { slopes, ratioX, leftOffset } = store;
+  console.log('slopes:', { slopes, ratioX, leftOffset, width, height, store }, 'params and store');
+
+  const { maxGradient, slopesRange } = slopesValues(store);
+  const { maxPosition } = maxPositionValues(store);
+
+  if (!slopes || slopes.length === 0) {
+    console.error('Slopes data is missing or empty.');
+    return;
+  }
+
+  clearCanvas(ctx, width, height);
+
+  ctx.save();
+  ctx.translate(leftOffset, 0);
+
+  // Calculate total height available for drawing, excluding the margins
+  const availableHeight = height - MARGIN_TOP - MARGIN_BOTTOM - 30;
+
+  // Calculate the vertical center of the chart
+  const centerY = MARGIN_TOP + 30 + availableHeight / 2;
+
+  console.log('centerY:', centerY, 'maxGradient:', maxGradient);
+
+  ctx.fillStyle = '#CFDDCE';
```

Comment:
store the color in a variable whose name indicates the element to which the color is to be applied. For example
`SLOPE_FILL_COLOR=#CFDDCE`